Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -883,23 +960,43 @@ private OrderResponse HandleCancelOrderRequest(CancelOrderRequest request)
 
         private void HandleOrderEvent(OrderEvent fill)
         {
-            // update the order status
-            var order = GetOrderByIdInternal(fill.OrderId);
-            if (order == null)
+            Order order;
+            OrderTicket ticket;
+            if (fill.Status.IsClosed())
             {
-                Log.Error("BrokerageTransactionHandler.HandleOrderEvent(): Unable to locate Order with id " + fill.OrderId);
-                return;
+                if(_openOrders.TryRemove(fill.OrderId, out order)) {
+                    _closedOrders.TryAdd(order.Id, order);
+                }
+                else {
+                    Log.Error("BrokerageTransactionHandler.HandleOrderEvent(): Unable to locate open Order with id " + fill.OrderId);
+                    return;
+                }
+
+                if (_openOrderTickets.TryRemove(order.Id, out ticket)) {
+                    _closedOrderTickets.TryAdd(order.Id, ticket);
+                }
+                else {
```

Comment:
The code style for the project places braces on their own lines and always includes braces for control structures.

Reply:
My bad, I'll change it